### PR TITLE
fix: env variables check blocks many AWS authentication methods 

### PR DESCRIPTION
**Pull request by fvdnabee**
### Description

Enforcing the specification of the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env variables blocks other authentication methods supported by the AWS go SDK from being used by argocd-vault-plugin. Even within env vars authentication, there exist other options than just `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`: e.g. using an assumed role on ec2/EKS. This PR removes the unnecessary check for the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env vars. It does remain recommended to set the proper AWS region via an environment variables however. 

**Fixes:** https://github.com/IBM/argocd-vault-plugin/issues/148

### Checklist
Please make sure that your PR fulfills the following requirements:
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] Tests for the changes have been updated
- [x] Docs have been added / updated

### Type of Change

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### Other information

Note, this PR was tested successfully with an IAM role for service accounts as the auth method on an AWS EKS cluster. In this test the following env variables were set by the EKS Pod Identity Webhook:
```
AWS_ROLE_ARN=arn:aws:iam::xxx:role/argocd-repo-server-secretsmanager-my-cluster
AWS_WEB_IDENTITY_TOKEN_FILE=/var/run/secrets/eks.amazonaws.com/serviceaccount/token
AWS_DEFAULT_REGION=eu-west-1
AWS_REGION=eu-west-1
```

Note, when no credentials are found the user is shown the following error after this PR is applied:
```
AVP_TYPE=awssecretsmanager ./bin/argocd-vault-plugin_1.2.0dev_linux_amd64 generate /tmp/secret.yaml
Error: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
Usage:
  argocd-vault-plugin generate <path> [flags]

Flags:
  -c, --config-path string   path to a file containing Vault configuration (YAML, JSON, envfile) to use
  -h, --help                 help for generate
  -s, --secret-name string   name of a Kubernetes Secret containing Vault configuration data in the argocd namespace of your ArgoCD host (Only available when used in ArgoCD)

NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
```

### CODE REVIEW

**Comment by werne2j** [comment]
@fvdnabee thanks for the PR! Can you please rebase your branch onto main?

**Comment by fvdnabee** [comment]
> @fvdnabee thanks for the PR! Can you please rebase your branch onto main?

Done!

**Comment by werne2j** [comment]
@fvdnabee Everything looks good, could you add a test or two to cover the new logic?

**Comment by fvdnabee** [comment]
@werne2j checking the error value of `session.NewSession` is of limited use as the aws sdk doesn't actually validate credentials at that time. You'll only know whether your credentials are valid and whether they are authorized  when performing API calls (at least this is my understanding). 

That said, I've added a test for a corner case in bc1982f which is covered by the SDK. Another test case demonstrates web identity federation authentication.

**Review by werne2j — APPROVED** [review]
Looks good!

File: pkg/config/config.go [hunk 1]
```diff
@@ -109,16 +109,13 @@ func New(v *viper.Viper, co *Options) (*Config, error) {
...
 		{
-			if !v.IsSet(types.EnvAWSAccessKey) || !v.IsSet(types.EnvAWSSecretAccessKey) {
-				return nil, fmt.Errorf("Must provide %s and %s for backend type %s",
-					types.EnvAWSAccessKey,
-					types.EnvAWSSecretAccessKey,
-					types.AWSSecretsManagerbackend,
-				)
-			}
-			s := session.Must(session.NewSession(&aws.Config{
+			s, err := session.NewSession(&aws.Config{
```
Comment (jkayani) [inline pkg/config/config.go:112]: This looks fine. The only problem here is that if someone is setting their AVP config through a k8s Secret, or a config file, then specifying the `AWS_BLAH` parameters will not get sent to the AWS SDK, and it won't. Perhaps you can add some code to loop through the Viper settings and set all `AWS_` config keys as env variables like we do for `VAULT_`: https://github.com/IBM/argocd-vault-plugin/blob/main/pkg/config/config.go#L159

File: pkg/types/constants.go [hunk 2]
```diff
@@ -2,20 +2,18 @@ package types
...
+	EnvAvpSecretID     = "AVP_SECRET_ID"
+	EnvAvpAuthType     = "AVP_AUTH_TYPE"
+	EnvAvpGithubToken  = "AVP_GITHUB_TOKEN"
+	EnvAvpK8sRole      = "AVP_K8S_ROLE"
+	EnvAvpK8sMountPath = "AVP_K8S_MOUNT_PATH"
+	EnvAvpK8sTokenPath = "AVP_K8S_TOKEN_PATH"
+	EnvAvpIBMAPIKey    = "AVP_IBM_API_KEY"
+	EnvAvpKvVersion    = "AVP_KV_VERSION"
+	EnvAvpPathPrefix   = "AVP_PATH_PREFIX"
+	EnvAWSRegion       = "AWS_REGION"
```
Comment (jkayani) [inline pkg/types/constants.go:16]: If you decide to take the other suggestion then you can drop this variable too

**Review by jkayani — COMMENTED** [review]
LGTM, but this change will only apply to AVP config from env variables. I'd prefer to have it work regardless of source of configuration, thus my comments. Let me know your thoughts

File: pkg/config/config.go (hunk 1 above)
Comment (fvdnabee) [reply pkg/config/config.go:112]: Unsure of the use-cases here as I wouldn't add AWS credentials to an AVP config file/secret on k8s, preferring IAM auth or envFrom from a dedicated AWS secret instead. But others might see a use for it, so I'll add support for setting envs vars for each `AWS_` entry in the config file.

File: pkg/types/constants.go (hunk 2 above)
Comment (fvdnabee) [reply pkg/types/constants.go:16]: For most auth methods it's important to ensure a valid region is set though. I believe an env variable is the most common way of doing so. In light of this I'd  suggest to keep EnvAWSRegion as a dedicated constant, see also PR #155 .

Comment (jkayani) [reply pkg/types/constants.go:16]: I'm convinced

File: pkg/config/config.go (hunk 1 above)
Comment (fvdnabee) [reply pkg/config/config.go:112]: @jkayani I just pushed e846f41 with a unit test for `AWS_` config entries. Hope I understood you right.

Comment (fvdnabee) [reply pkg/config/config.go:112]: Note, the fake credentials are taken from https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-set

**Review by jkayani — APPROVED** [review]
Exactly what I wanted to see, thanks a lot @fvdnabee !
